se agrega el R Dev Day (#68)
LatinR/web-LatinR2024: cronograma/r_dev_day/RDevDay-pt.qmd

---
title: "R Dev Day"
knitr:
  opts_chunk:
    echo: false
    message: false
    warning: false
editor: 
  markdown: 
    wrap: sentence
---

# R Dev Day \@ LatinR 2024

**Datas:** 18 e 19 de novembro de 2024, das 17h30 às 20h (UTC-3)

**Modo:** Online

**Inscrição:** <https://pretix.eu/r-contributors/r-dev-day-latinr2024/>

**Idiomas:** Inglês, português e espanhol

# Sobre

> O R Dev Day \@ LatinR 2024 é um evento colaborativo no qual os participantes trabalharão em pequenos grupos para contribuir para o R básico ou para a infraestrutura que dá suporte a essas contribuições, por exemplo

Reviewer: ```suggestion
O R Dev Day \@ LatinR 2024 é um evento colaborativo no qual as pessoas participantes trabalharão em pequenos grupos para contribuir para o R base ou para a infraestrutura que dá suporte a essas contribuições, por exemplo:
```

> -   Traduzir strings de mensagens para espanhol/português brasileiro

Reviewer: ```suggestion
-   Traduzir textos de mensagens para espanhol/português brasileiro
```

> -   Propor melhorias nos arquivos de ajuda do R

Reviewer: ```suggestion
-   Propor melhorias nos arquivos de ajuda do R (conteúdo que acessamos no `help` ou `?` no R)
```

Reviewer 2: Apply similar update to English:

-   Propose improvements to the R help files (that are accessed via `help` or `?` in R)

-   Analisar um erro relatado no R
-   Desenvolver uma correção para um erro relatado no R

> -   Adicionar conteúdo ao [R Dev Guide](https://contributor.r-project.org/rdevguide/)

Reviewer: ```suggestion
-   Adicionar conteúdo ao [R Dev Guide](https://contributor.r-project.org/rdevguide/), o guia de contribuição ao R base
```

Reviewer 2: Apply similar update to English:

-   Add content to the [R Dev Guide](https://contributor.r-project.org/rdevguide/), the guide for contributors to base R

> -   Melhorar o [R Translations Dashboard](https://contributor.r-project.org/translations-dashboard/)

Reviewer: ```suggestion
-   Melhorar o [R Translations Dashboard](https://contributor.r-project.org/translations-dashboard/), o painel utilizado para acompanhar o andamento das traduções 
```

Reviewer 2: Apply similar update to English:

- Improve the [R Translations Dashboard](https://contributor.r-project.org/translations-dashboard/), the dashboard used to monitor the status of translations

> Tanto colaboradores novos quanto experientes estão convidados a participar!

Reviewer: ```suggestion
Tanto pessoas colaboradoras novas quanto experientes estão convidadas a participar!
```

## Quando

Haverá duas sessões na segunda-feira, 18 de novembro, e na terça-feira, 19 de novembro. No mesmo horário todos os dias:

> | UTC           | Buenos Aires (UTC-3) | Hermosillo (UTC-7) |

Reviewer: ```suggestion
| UTC           | Buenos Aires e Brasília (UTC-3) | Hermosillo (UTC-7) |
```

Reviewer 2: Copy over to English and Spanish translation too

|:--------------|:---------------------|:-------------------|
| 8:30pm - 11pm | 5:30pm - 8pm         | 1:30pm - 4pm       |

Recomendamos que você participe de ambas as sessões, se possível, mas não há problema em participar de apenas uma sessão.

## Comunicação

Haverá uma breve introdução (\~10 minutos) no início de cada sessão em inglês, espanhol e português.

Usaremos o canal "r-dev-day" no Slack do LatinR para discussões relacionadas - o uso de inglês, espanhol ou português é bem-vindo.

> Os pequenos grupos podem acordar um idioma comum para trabalhar - o inglês será o padrão, mas os facilitadores que falam espanhol e português estarão disponíveis para ajudar.

Reviewer: ```suggestion
Os grupos menores podem acordar um idioma comum para trabalhar - o inglês será o padrão, mas as pessoas facilitadoras que falam espanhol e português estarão disponíveis para ajudar.
```

## Tarefas

> As tarefas serão preparadas como [problemas neste repositório com o rótulo `LatinR2024`](https://github.com/r-devel/r-dev-day/issues?q=is%3Aissue+is%3Aopen+label%3ALatinR2024). Você pode adicionar problemas ou discutir ideias para problemas, conforme descrito no [README](https://github.com/r-devel/r-dev-day/blob/main/README.md).

Reviewer: ```suggestion
As tarefas serão preparadas como [*issues* neste repositório com o rótulo `LatinR2024`](https://github.com/r-devel/r-dev-day/issues?q=is%3Aissue+is%3Aopen+label%3ALatinR2024). Você pode adicionar *issues* ou discutir ideias para *issues*, conforme descrito no [README](https://github.com/r-devel/r-dev-day/blob/main/README.md).
```

> Os problemas estão resumidos na planilha do [R Dev Day \@ LatinR 2024 Working Group Allocations](https://docs.google.com/spreadsheets/d/1Nuwusm7Xgwpw54KZnnrnNVT9vPkOk2qmromBEbYNHnU/edit). Adicione seu nome à lista de colaboradores se estiver interessado em trabalhar em um problema. Você pode se incluir em várias questões, pois não se trata de um compromisso final - as alocações serão finalizadas no início de cada sessão.

Reviewer: ```suggestion
As *issues* estão resumidos na planilha do [R Dev Day \@ LatinR 2024 Working Group Allocations](https://docs.google.com/spreadsheets/d/1Nuwusm7Xgwpw54KZnnrnNVT9vPkOk2qmromBEbYNHnU/edit). Adicione seu nome à lista de pessoas colaboradoras se tiver interesse em trabalhar em uma *issue*. Você pode se incluir em várias *issues*, pois não se trata de um compromisso final - as alocações serão finalizadas no início de cada sessão.
```

> Fique de olho nas edições antes do evento. Talvez você precise de preparação adicional para trabalhar em determinadas questões, por exemplo, aprender a criar o R no macOS ou ler um material de apoio.

Reviewer: ```suggestion
Fique de olho nas edições antes do evento. Talvez você precise de preparação adicional para trabalhar em determinadas questões, por exemplo, aprender a fazer o *build* do R no macOS ou ler um material de apoio.
```

## Configuração do desenvolvimento

> A maioria das pessoas poderá trabalhar somente pelo navegador (usando o Weblate, a interface do navegador do GitHub, o GitHub Codespaces ou o Gitpod). O trabalho com bugs específicos do sistema operacional pode exigir o uso ou a criação do R em seu próprio computador.

Reviewer: ```suggestion
A maioria das pessoas poderá trabalhar somente pelo navegador (usando o Weblate, a interface do navegador do GitHub, o GitHub Codespaces ou o Gitpod). O trabalho com bugs específicos do sistema operacional pode exigir fazer o *build* do R no seu próprio computador.
```

## Preparação

Algumas coisas que você pode fazer para ajudar a se preparar, se tiver tempo:

-   [Solicite uma conta no Bugzilla](https://contributor.r-project.org/rdevguide/IssueTrack.html#bugzilla-account) do R se você quiser trabalhar em bugs no código/documentação e ainda não tiver uma conta. O processo pode levar alguns dias, mas é apenas um "é bom ter".
-   [Crie uma conta no Weblate](https://translate.rx.studio/projects/r-project/) se você quiser trabalhar com traduções. Isso também pode ser feito durante o evento.

> -   Se quiser trabalhar em bugs no código do R (em comparação com a documentação, o R Translations Dashboard ou o R Dev Guide) e não se sentir confortável com a criação do R a partir do código-fonte, recomendamos que aprenda a usar o R Dev Container, conforme descrito na seção [Outros recursos](#outros-recursos) (leva aproximadamente 1 hora). Isso não será necessário para todas as tarefas que envolvem código, mas abrirá mais possibilidades.

Reviewer: ```suggestion
-   Se quiser trabalhar em bugs no código do R (em comparação com a documentação, o *R Translations Dashboard* ou o *R Dev Guide*) e não se sentir confortável com fazer o *build* do R a partir do código-fonte, recomendamos que aprenda a usar o *R Dev Container*, conforme descrito na seção [Outros recursos](#outros-recursos) (leva aproximadamente 1 hora). Isso não será necessário para todas as tarefas que envolvem código, mas abrirá mais possibilidades.
```

## Outros recursos {#outros-recursos}

[Reviewing Bug Reports](https://blog.r-project.org/2019/10/09/r-can-use-your-help-reviewing-bug-reports/index.html), postagem do blog do R (leitura de 3 minutos)

[Debugging in R](https://youtu.be/31kl3Y0MUWc) (1h45 de vídeo)

[Analysing Bugs/Contributing Patches](https://contributor.r-project.org/tutorials/contributing-to-r/)(2h15 de vídeo + exercícios - pressupõe algum conhecimento de depuração em R).

[Starting Workspace](https://contributor.r-project.org/r-dev-env/container_setup/gitpod_workspace/workspacestart/) e [Stopping and restarting](https://contributor.r-project.org/r-dev-env/container_setup/gitpod_workspace/workspacestop_and_restart/), para aprender a usar o R Dev Container por meio do navegador com espaços de trabalho do Gitpod, seguido por [Running R](https://contributor.r-project.org/r-dev-env/tutorials/running_r/), [Building R](https://contributor.r-project.org/r-dev-env/tutorials/building_r/) e [R Contribution Workflow](https://contributor.r-project.org/r-dev-env/tutorials/contribution_workflow/) para aprender os fundamentos do uso do R Dev Container para desenvolver alterações no R básico (guia passo a passo de \<1 hora).

> [R Dev Guide](https://contributor.r-project.org/rdevguide/) (ebook). Foi criado como uma referência para colaboradores sobre diferentes aspectos da contribuição para o R. Em particular, consulte o Capítulo 2 para saber como criar o R em seu próprio computador. Atualmente, as instruções para macOS não foram incorporadas, mas há uma [versão preliminar](https://github.com/r-devel/rcwg/blob/main/working_documents/install_r_macos.md).

Reviewer: ```suggestion
[R Dev Guide](https://contributor.r-project.org/rdevguide/) (ebook). Foi criado como uma referência para pessoas colaboradoras sobre diferentes aspectos da contribuição para o R. Em particular, consulte o Capítulo 2 para saber como fazer o *build* do R no seu próprio computador. Atualmente, as instruções para macOS não foram incorporadas, mas há uma [versão preliminar](https://github.com/r-devel/rcwg/blob/main/working_documents/install_r_macos.md).
```
